src/Common.cpp

```diff
@@ -5,328 +5,9 @@
-{
-	const auto acc_op = [](const std::size_t& a, const std::string& b) noexcept->std::size_t { return a + std::size_t{1u} + b.size(); };
-	auto total_size = std::accumulate(std::begin(strs), std::end(strs), std::size_t{0u}, acc_op);
-	std::string result;
-	result.reserve(total_size);
-
-	for (auto iter = std::begin(strs); iter != std::end(strs); ++iter)
-	{
-		result += (*iter);
-		if (iter + 1 != std::end(strs))
-		{
-			result.push_back(delim);
-		}
-	}
-
-	result.shrink_to_fit();
-	return result;
-}
-std::string NAS2D::joinSkipEmpty(std::vector<std::string> strs)
-{
-	const auto acc_op = [](const std::size_t& a, const std::string& b) noexcept->std::size_t { return a + b.size(); };
-	auto total_size = std::accumulate(std::begin(strs), std::end(strs), std::size_t{0u}, acc_op);
-	std::string result;
-	result.reserve(total_size);
-	for (const auto& s : strs)
-	{
-		if (s.empty()) { continue; }
-		result += s;
-	}
-	result.shrink_to_fit();
-	return result;
-}
-
-std::string NAS2D::joinSkipEmpty(std::vector<std::string> strs, char delim)
-{
-	const auto acc_op = [](const std::size_t& a, const std::string& b) noexcept->std::size_t { return a + std::size_t{1u} + b.size(); };
-	auto total_size = std::accumulate(std::begin(strs), std::end(strs), std::size_t{0u}, acc_op);
-	std::string result;
-	result.reserve(total_size);
-
-	for (auto iter = std::begin(strs); iter != std::end(strs); ++iter)
-	{
-		if ((*iter).empty()) { continue; }
-		result += (*iter);
-		if (iter + 1 != std::end(strs))
-		{
-			result.push_back(delim);
-		}
-	}
-
-	result.shrink_to_fit();
-	return result;
-}
-
-std::string NAS2D::trimWhitespace(std::string string)
-{
-	const auto first_non_space = string.find_first_not_of(" \r\n\t\v\f");
-	if (first_non_space == std::string::npos)
-	{
-		return std::string{};
-	}
-	const auto last_non_space = string.find_last_not_of(" \r\n\t\v\f");
-	return string.substr(first_non_space, last_non_space - first_non_space + 1);
-}
-
-bool NAS2D::startsWith(std::string_view string, std::string_view start) noexcept
-{
-	return string.compare(0, start.size(), start) == 0;
-}
-
-bool NAS2D::endsWith(std::string_view string, std::string_view end) noexcept
-{
-	return string.compare(string.size() - end.size(), end.size(), end) == 0;
-}
-
-bool NAS2D::startsWith(std::string_view string, char start) noexcept
-{
-	return !string.empty() && string.front() == start;
-}
-
-bool NAS2D::endsWith(std::string_view string, char end) noexcept
-{
-	return !string.empty() && string.back() == end;
-}
-
-/**
- * \fn int divideUp(int a, int b)
- *
- * Basic integer division that rounds up to the nearest whole number.
- *
- * \param	to_divide	Number to be divided.
- * \param	divisor		Divisor.
- *
- * \return	Returns the divided number rounded up to the nearest whole number.
- */
-int NAS2D::divideUp(int to_divide, int divisor)
-{
-	return (to_divide + (divisor - 1)) / divisor;
-}
+#include "MathUtils.cpp"
+#include "StringUtils.cpp"
```

Comment:
I don't think this is required in a .cpp file. Preserving the header and having header includes for the split files should be enough to preserve compatibility. It shouldn't be needed at the .cpp file level.

As long as the methods are defined in some .cpp file that is part of the project, the linker should be able to find the object code and link them. Actually, including source files in another source file could potentially cause link errors, as the same object methods are defined in two different compilation units with the same name. Maybe you'll get lucky and have the linker discard the duplicates, assuming they are exactly the same, though best not to rely on that behavior.

Reply:
Fixed. 

Yeah that was a brain-fart. I dunno why I did that. :P